decoder.js

```diff
@@ -33,15 +43,94 @@ Decoder.prototype._writeOffset = function (chunk, offset, cb) {
     if (this._inBody) {
       offset = this._writeBody(chunk, offset, cb)
       if (offset === 0) return // chunk not consumed - _writeOffset will be called again when ready
-    } else {
+    } else if (this._inHead) {
       offset = this._writeHead(chunk, offset)
+    } else if (this._inInterleavedHead) {
+      offset = this._writeInterleavedHead(chunk, offset)
+    } else if (this._inInterleavedPacket) {
+      offset = this._writeInterleavedPacket(chunk, offset, cb)
+    } else {
+      if (chunk[offset] === INTERLEAVED_SIGN) {
+        offset = this._writeInterleavedHead(chunk, offset)
+      } else {
+        offset = this._writeHead(chunk, offset)
+      }
     }
   }
   cb()
 }
+Decoder.prototype._writeInterleavedHead = function (chunk, offset) {
+  if (this._interleavedHeaderOffset === 0) {
+    debug('start of interleaved header')
+    this._inInterleavedHead = true
+  }
+  if (chunk.length - offset < 4) {
```

Comment:
Shouldn't the `4` here be `INTERLEAVED_HEADER_BYTES` if I understand the code correctly?